doozerlib/config.py

```diff
@@ -76,39 +76,45 @@ def update(self, key, val):
         for rpm in self.runtime.rpm_metas():
             self.update_meta(rpm, key, val)
 
-    def config_print(self, key=None, name_only=False):
+    def config_print(self, key=None, name_only=False, as_yaml=False):
         """
         Print name, sub-key, or entire config
         """
-        def _do_print(meta, k):
-            if name_only:
-                print(meta.config_filename)
-            else:
-                if k:
-                    val = meta.config.get(k, None)
-                else:
-                    val = meta.config.primitive()
 
-                val = yaml.safe_dump(val, default_flow_style=False)
+        data = dict(images={}, rpms={})
 
-                print("*****" + meta.config_filename + "*****")
-                print(val)
-                print('')
+        def _collect_data(metas, output):
+            for meta in metas:
+                if name_only:
+                    entry = meta.config_filename
+                elif key:
+                    entry = meta.config.get(key, None)
+                else:
+                    entry = meta.config.primitive()
+                output[meta.config_filename.replace('.yml', '')] = entry
 
-        image_metas = self.runtime.image_metas()
-        rpm_metas = self.runtime.rpm_metas()
+        _collect_data(self.runtime.image_metas(), data["images"])
+        _collect_data(self.runtime.rpm_metas(), data["rpms"])
 
-        if image_metas:
-            print('')
-            print('********* Images *********')
-            for img in image_metas:
-                _do_print(img, key)
+        if as_yaml:
+            print yaml.safe_dump(data, default_flow_style=False)
```

Comment:
needs ()
Though in general, I feel these should be switched to our `*_print` functions as they use click print which is safer.
But that's, I admit, out of the context of this PR as I was the one that did that in the first place :P

Reply:
works without the parens, but sure, consistency.
also feel free to go through and convert all these to a proper print...